app/products/agents/components/controls_bar/index.tsx

```diff
@@ -0,0 +1,139 @@
+            }),
+            [
+                {
+                    text: intl.formatMessage({
+                        id: 'agents.regenerate.cancel',
+                        defaultMessage: 'Cancel',
+                    }),
+                    style: 'cancel',
+                },
+                {
+                    text: intl.formatMessage({
+                        id: 'agents.regenerate.confirm',
+                        defaultMessage: 'Regenerate',
+                    }),
+                    onPress: onRegenerate,
+                    style: 'destructive',
+                },
+            ],
+        );
+    }, [intl, onRegenerate]);
+
+    if (!showStopButton && !showRegenerateButton) {
+        return null;
+    }
+
+    return (
+        <View style={styles.container}>
+            {showStopButton && (
+                <TouchableOpacity
+                    onPress={preventDoubleTap(onStop)}
+                    style={styles.button}
+                    activeOpacity={0.7}
+                    testID='agents.controls_bar.stop_button'
+                >
+                    <CompassIcon
+                        name='close-circle-outline'
+                        size={14}
+                        color={changeOpacity(theme.centerChannelColor, 0.64)}
+                    />
+                    <FormattedText
+                        id='agents.controls.stop'
+                        defaultMessage='Stop Generating'
+                        style={styles.buttonText}
+                    />
+                </TouchableOpacity>
+            )}
+            {showRegenerateButton && (
+                <TouchableOpacity
+                    onPress={preventDoubleTap(handleRegenerate)}
+                    style={styles.button}
+                    activeOpacity={0.7}
+                    testID='agents.controls_bar.regenerate_button'
+                >
+                    <CompassIcon
+                        name='refresh'
+                        size={14}
+                        color={changeOpacity(theme.centerChannelColor, 0.64)}
+                    />
+                    <FormattedText
+                        id='agents.controls.regenerate'
+                        defaultMessage='Regenerate'
+                        style={styles.buttonText}
+                    />
+                </TouchableOpacity>
+            )}
+        </View>
+    );
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return StyleSheet.create({
+        container: {
+            flexDirection: 'row',
+            gap: 8,
+            marginTop: 8,
+        },
+        button: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            gap: 6,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            borderRadius: 4,
+            paddingVertical: 8,
+            paddingHorizontal: 10,
+            minHeight: 44, // Touch-optimized height
```

Comment:
And that other constant that I mention, perhaps make it available in a constant file so you can use it here too